spinflip/diagonalizer.py

```diff
@@ -49,22 +44,29 @@ def timescales(self, minmax = True):
         else:
             return hbar/differences
         
-    def state_evolution_plotter(self, t_lim = 'timescale', resolution=500, quantity = 'state_right', ylim = None, init_array = np.diag((1,0,0,0,0,0)), savefig = False):
+        #time array of neutrino density matrix, with init_array as intial condition
+    def state_evolution(self, resolution, t_lim, init_array):
         if t_lim == 'timescale':
            t_lim = self.timescale
            print('Largest timescale = '+str(t_lim)+ ' s')
+        return np.array([self.U_flavor(t) @ init_array @ np.linalg.inv(self.U_flavor(t)) for t in np.linspace(0, t_lim, resolution)])
+    
+    #calculates the largest component of initial_ket_f and returns the component and the corresponding eigenvector
+    def largest_ket_component(self, init_ket_f):
```

Comment:
Could you add a comment saying that f_to_e contains the eigenvectors of H, where the first index is in the energy basis and the second is in the flavor basis (i.e., the first specifies which eigenvector and the second specifies the component in flavor space)

Reply:
And say components is the list of the dot products of the initial state with each eigenvector. It's clear after looking at it for a bit, but not immediately obvious, given that I have to recall indexing conventions

Reply:
is it possible for you edit my commits directly? I'm not totally sure what these are referring to - maybe the lines github is showing 52 thru 55, are pointing to the wrong place